hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1303,7 +1303,7 @@ public void rename(final Path source, final Path destination,
         if (isAtomicRenameKey(source.toUri().getPath())) {
           LOG.debug("source dir {} is an atomicRenameKey",
               source.toUri().getPath());
-          renameAtomicityUtils.preRename(srcBlobProperties);
+          renameAtomicityUtils.preRename(srcBlobProperties, isCreateOnBlobEndpoint());
```

Comment:
Since we are adding knobs that create can be done on DFS endpoint, although we intend to rename over Blob.

So, this flag gives the information if ingress is not happening over DFS.

Reply:
`isAtomicRenameKey() and calls to renameAtomocityUtils should happen only in BlobEndpoint case.`

correct. By ingressConfig, we mean that create, append, flush would be done over dfs endpoint. If endpoint in the config is BLOB or blobEnableConfig is on, the rename will go through the new added flow (the blob related code), but the create of RenameJSON will happen over dfs because of ingressConfig.